protocols/deflate/tests/test.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use futures::prelude::*;
+use libp2p_core::transport::{ListenerEvent, Transport};
+use libp2p_core::upgrade::Negotiated;
+use libp2p_deflate::{DeflateConfig, DeflateOutput};
+use libp2p_tcp::{TcpConfig, TcpTransStream};
+use log::info;
+use quickcheck::QuickCheck;
+use tokio::{self, io};
+
+#[test]
+fn deflate() {
+    let _ = env_logger::try_init();
+
+    fn prop(message: Vec<u8>) -> bool {
+        let server_transport = TcpConfig::new().with_upgrade(DeflateConfig {});
+        let client_transport = TcpConfig::new().with_upgrade(DeflateConfig {});
+        run(server_transport, client_transport, message);
+        true
+    }
+
+    QuickCheck::new()
```

Comment:
I'm not sure if there's any need for using this, I copied it from another protocol test.